codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2DataFrame.java

```diff
@@ -73,11 +75,12 @@ public DefaultHttp2DataFrame(ByteBuf content, boolean endStream, int padding) {
         this.endStream = endStream;
         verifyPadding(padding);
         this.padding = padding;
+        initialFlowControlledBytes = content().readableBytes() + padding;
```

Comment:
a frame cannot be larger than `2^24-1` [1][2] and window update only supports 31 bits [3]. So this should only be an issue if a user creates an outbound data frame and supplies more data than a single frame can handle. The underlying connection API will take care to chunk the data/padding appropriately in this case, but there are other APIs (e.g`FlowControlledFrameSizeEstimator#size(msg)` return type is `int`) so if this method returned a `long` there maybe mis-alignment. This case seems really unlikely so why don't we just throw if there is overflow and keep the `int` interface. We can provide utilities later (`static Http2DataFrame[] frames(ByteBuf content, int padding)`) to split the frame objects up if necessary.

[1] https://tools.ietf.org/html/rfc7540#section-4.1
[2] https://tools.ietf.org/html/rfc7540#section-4.2
[3] https://tools.ietf.org/html/rfc7540#section-6.9